metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -114,56 +114,86 @@ final class MultilineStringFormattingProvider(
       endPos: meta.Position,
-      sourceText: String
+      sourceText: String,
+      newlineAdded: Boolean
   ): Boolean = {
-    var tokenIndex = 0
-    var stringFound = false
-    var shouldAddPipes = false
-    while (!stringFound && tokenIndex < tokens.size) {
-      tokens(tokenIndex) match {
-        case token: Constant.String if inToken(startPos, endPos, token) =>
-          stringFound = true
-          shouldAddPipes = isMultilineString(sourceText, token) &&
-            hasStripMarginSuffix(tokenIndex, tokens)
-        case start: Interpolation.Start if start.start < startPos.start =>
-          var endIndex = tokenIndex + 1
+    if (pipeInScope(startPos, endPos, sourceText, newlineAdded)) {
+      def shouldFormatMultiString: PartialFunction[(Token, Int), Boolean] = {
+        case (token: Constant.String, index: Int) =>
+          isMultilineString(sourceText, token) && hasStripMarginSuffix(
+            index,
+            tokens
+          ) && inToken(startPos, endPos, token)
+      }
+
+      def shouldFormatInterpolationString
+          : PartialFunction[(Token, Int), Boolean] = {
+        case (token: Interpolation.Start, index: Int)
+            if token.start < startPos.start => {
+          var endIndex = index + 1
           while (!tokens(endIndex)
               .isInstanceOf[Interpolation.End]) endIndex += 1
-          val end = tokens(endIndex)
-          stringFound = end.end > startPos.end
-          shouldAddPipes =
-            stringFound && isMultilineString(sourceText, start) &&
-              hasStripMarginSuffix(endIndex, tokens)
-        case _ =>
+          isMultilineString(sourceText, token) && hasStripMarginSuffix(
+            endIndex,
+            tokens
+          )
+        }
       }
-      tokenIndex += 1
-    }
-    shouldAddPipes
+      tokens.zipWithIndex
+        .exists(
+          shouldFormatMultiString orElse shouldFormatInterpolationString orElse {
+            case _ => false
+          }
+        )
+    } else false
   }
 
-  private def withToken(
-      textId: TextDocumentIdentifier,
-      range: Range,
-      newlineAdded: Boolean
-  )(fn: (String, meta.Position) => List[TextEdit]): Future[List[TextEdit]] =
-    Future {
-      val source = textId.getUri.toAbsolutePath
-      if (source.exists) {
-        val sourceText = buffer.get(source).getOrElse("")
-        val virtualFile = Input.VirtualFile(source.toString(), sourceText)
-        val startPos = range.getStart.toMeta(virtualFile)
-        val endPos = range.getEnd.toMeta(virtualFile)
-        if (pipeInScope(startPos, endPos, sourceText, newlineAdded)) {
-          val tokens =
-            virtualFile.tokenize.toOption
-          tokens
-            .filter(multilineStringInTokens(_, startPos, endPos, sourceText))
-            .map(_ => fn(sourceText, startPos))
-            .getOrElse(Nil)
-        } else Nil
-      } else Nil
-    }
+  private def withTokenOpt(textId: TextDocumentIdentifier, range: Range)(
```

Comment:
I don't think we need to change the name to `withTokenOpt` since we don't have any other method like that.